src/charm.py

```diff
@@ -129,6 +131,19 @@ def _on_start(self, event: EventBase) -> None:
         self.cluster.relation.data[self.unit].update(unit_config)
         self.cluster.relation.data[self.unit].update({"state": "started"})
 
+    def _on_config_changed(self, event):
+        """Handler for the `config-changed` event.
+
+        This includes:
+            - Writing config to config files\
+            - Restarting zookeeper service
+        """
+        self.snap.write_properties(
+            properties=self.zookeeper_config.create_properties(self.config),
+            property_label="zookeeper",
+        )
+        self.snap.restart_snap_service("zookeeper")
```

Comment:
To allow for a rolling restart:
```suggestion
        self.on[self.restart.name].acquire_lock.emit()
```